src/auth.js

```diff
@@ -66,6 +67,25 @@ const storeApiKey = (key) => {
   return null
 }
 
+const getStateObject = (key) => {
+  let state = {}
+  if (window.localStorage) {
+    const stringifiedState = window.localStorage.getItem(key)
+    state = stringifiedState ? JSON.parse(stringifiedState) : state
+  }
+  return state
+}
+
+const storeStateObject = (state) => {
+  if (window.localStorage) {
+    const key = uuidv1()
+    const stringifiedState = JSON.stringify(state)
+    window.localStorage.setItem(key, stringifiedState)
+    return key
+  }
+  return null
```

Comment:
I'm not sure if `window.localStorate` will every be falsy. However, if it is, then the most sensible thing to do here would be to return `state` as-is, that way it can still be passed to OAuth which should work in most cases.